### PR TITLE
PI-2365: Community API proxy

### PR DESCRIPTION
**Pull request by pmcphee77**

### CODE REVIEW

File: projects/court-case-and-delius/deploy/values-dev.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ generic-service:
     SPRING_DATASOURCE_HIKARI_MINIMUMIDLE: 0
 
     LOGGING_LEVEL_UK_GOV_DIGITAL_JUSTICE_HMPPS: DEBUG
+    COMMUNITY-API_URL: https://community-api-secure.test.delius.probation.hmpps.dsd.io
```
Comment (marcus-bcl) [inline projects/court-case-and-delius/deploy/values-dev.yml:18]: Minor thing - but we're better off using these URLs for Community API:
* https://community-api.test.probation.service.justice.gov.uk
* https://community-api.pre-prod.delius.probation.hmpps.dsd.io
* https://community-api.probation.service.justice.gov.uk

They point to the same service so it doesn't matter too much, but people were meant to migrate to the new URLs a while ago.

**Review by marcus-bcl — APPROVED** [review]
Looks great, nice and simple!

File: projects/court-case-and-delius/deploy/values-dev.yml (hunk 1 above)
Comment (pmcphee77) [reply projects/court-case-and-delius/deploy/values-dev.yml]: made the changes

File: projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @GetMapping
+    fun proxy(request: HttpServletRequest, response: HttpServletResponse): ResponseEntity<String> {
+
+        val headerNames = request.headerNames
+        val headers = mutableMapOf<String, String>()
+        while (headerNames.hasMoreElements()) {
+            val headerName = headerNames.nextElement()
+            headers[headerName] = request.getHeader(headerName)
+        }
+        headers[HttpHeaders.CONTENT_TYPE] = MediaType.APPLICATION_JSON_VALUE
```
[the remark this answers is not in the document]
Comment (pmcphee77) [reply projects/court-case-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/proxy/CommunityApiController.kt]: Yep thats better. Changed

**Review by marcus-bcl — APPROVED** [review]

**Pull request merged**